src/libraries/Common/src/System/Text/Json/PooledByteBufferWriter.cs

```diff
@@ -1,249 +1,97 @@
-                sizeHint = Math.Max(sizeHint, MaximumBufferSize - currentLength);
-            }
-
-            if (sizeHint > availableSpace)
-            {
-                int growBy = Math.Max(sizeHint, currentLength);
-
-                int newSize = currentLength + growBy;
-
-                if ((uint)newSize > MaximumBufferSize)
-                {
-                    newSize = currentLength + sizeHint;
-                    if ((uint)newSize > MaximumBufferSize)
-                    {
-                        ThrowHelper.ThrowOutOfMemoryException_BufferMaximumSizeExceeded((uint)newSize);
-                    }
-                }
-
-                byte[] oldBuffer = _rentedBuffer;
-
-                _rentedBuffer = ArrayPool<byte>.Shared.Rent(newSize);
-
-                Debug.Assert(oldBuffer.Length >= _index);
-                Debug.Assert(_rentedBuffer.Length >= _index);
-
-                Span<byte> oldBufferAsSpan = oldBuffer.AsSpan(0, _index);
-                oldBufferAsSpan.CopyTo(_rentedBuffer);
-                oldBufferAsSpan.Clear();
-                ArrayPool<byte>.Shared.Return(oldBuffer);
-            }
-
-            Debug.Assert(_rentedBuffer.Length - _index > 0);
-            Debug.Assert(_rentedBuffer.Length - _index >= sizeHint);
-        }
-
         public override async ValueTask<FlushResult> FlushAsync(CancellationToken cancellationToken = default)
         {
             Debug.Assert(_stream is not null);
 #if NET
             await _stream.WriteAsync(WrittenMemory, cancellationToken).ConfigureAwait(false);
 #else
-            Debug.Assert(_rentedBuffer != null);
-            await _stream.WriteAsync(_rentedBuffer, 0, _index, cancellationToken).ConfigureAwait(false);
+            await _stream.WriteAsync(_buffer.DangerousGetUnderlyingBuffer(), _buffer.ActiveStartOffset, _buffer.ActiveLength, cancellationToken).ConfigureAwait(false);
 #endif
             Clear();
 
             return new FlushResult(isCanceled: false, isCompleted: false);
         }
 
         public override bool CanGetUnflushedBytes => true;
-        public override long UnflushedBytes => _index;
+        public override long UnflushedBytes => _buffer.ActiveLength;
 
         // This type is used internally in JsonSerializer to help buffer and flush bytes to the underlying Stream.
         // It's only pretending to be a PipeWriter and doesn't need Complete or CancelPendingFlush for the internal usage.
         public override void CancelPendingFlush() => throw new NotImplementedException();
         public override void Complete(Exception? exception = null) => throw new NotImplementedException();
     }
-
-    internal static partial class ThrowHelper
-    {
-        [DoesNotReturn]
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        public static void ThrowOutOfMemoryException_BufferMaximumSizeExceeded(uint capacity)
-        {
-            throw new OutOfMemoryException(SR.Format(SR.BufferMaximumSizeExceeded, capacity));
```

Comment:
Do we need to remove the resource string as well?

Reply:
The STJ one is still used, deleted the one in Logging.Console.